drivers/clock_control/clock_control_nrf2_lfclk.c

```diff
@@ -234,6 +236,45 @@ static int lfclk_init(const struct device *dev)
 		return -EIO;
 	}
 
+#if LFCLK_HAS_LFXO
+	int ret;
+
+	ret = lfosc_get_accuracy(&dev_data->max_accuracy);
+	if (ret < 0) {
+		return ret;
+	}
+
+	switch (NRF_APPLICATION_BICR_NS->LFOSC.MODE) {
+	case BICR_LFOSC_LFXOCONFIG_MODE_Crystal:
+		clock_options[LFCLK_NUM_OPTS - 2].accuracy = dev_data->max_accuracy;
+		clock_options[LFCLK_NUM_OPTS - 2].precision = 0;
+		clock_options[LFCLK_NUM_OPTS - 2].src = NRFS_CLOCK_SRC_LFCLK_XO_PIERCE;
+
+		clock_options[LFCLK_NUM_OPTS - 1].accuracy = dev_data->max_accuracy;
+		clock_options[LFCLK_NUM_OPTS - 1].precision = 1;
+		clock_options[LFCLK_NUM_OPTS - 1].src = NRFS_CLOCK_SRC_LFCLK_XO_PIERCE_HP;
+		break;
+	case BICR_LFOSC_LFXOCONFIG_MODE_ExtSine:
+		clock_options[LFCLK_NUM_OPTS - 2].accuracy = dev_data->max_accuracy;
+		clock_options[LFCLK_NUM_OPTS - 2].precision = 0;
+		clock_options[LFCLK_NUM_OPTS - 2].src = NRFS_CLOCK_SRC_LFCLK_XO_EXT_SINE;
+
+		clock_options[LFCLK_NUM_OPTS - 1].accuracy = dev_data->max_accuracy;
+		clock_options[LFCLK_NUM_OPTS - 1].precision = 1;
+		clock_options[LFCLK_NUM_OPTS - 1].src = NRFS_CLOCK_SRC_LFCLK_XO_EXT_SINE_HP;
+		break;
+	case BICR_LFOSC_LFXOCONFIG_MODE_ExtSquare:
+		clock_options[LFCLK_NUM_OPTS - 2].accuracy = dev_data->max_accuracy;
+		clock_options[LFCLK_NUM_OPTS - 2].precision = 0;
+		clock_options[LFCLK_NUM_OPTS - 2].src = NRFS_CLOCK_SRC_LFCLK_XO_EXT_SQUARE;
+		break;
```

Comment:
In this case, when `clock_options[LFCLK_NUM_OPTS - 1]` is empty, this loop: https://github.com/zephyrproject-rtos/zephyr/blob/dded562dc301c72757a123b6002f98fe74b36126/drivers/clock_control/clock_control_nrf2_lfclk.c#L164-L172 should not look into that item. I think we need `dev_data->clock_option_cnt` to use for that loop instead of `ARRAY_SIZE(clock_options)`.

Reply:
good point